dynamics.js

```diff
@@ -48,3 +48,201 @@ closerIcon.addEventListener('click', deactiveMenu);
+const popupCardButtonIcon1 = document.createElement('img');
+const popupCardButtonIcon2 = document.createElement('img');
+
+//  Add id to each element
+popupCloser.id = 'popupCloser';
+popupTittle.id = 'popupTittle';
+popupParagraph.id = 'popupParagraph';
+projectImage.id = 'projectImage';
+technologiesList.id = 'technologiesList';
+firstTechnology.id = 'firstTechnology';
+secondTechnology.id = 'secondTechnology';
+thirdTechnology.id = 'thirdTechnology';
+firstButton.id = 'firstButton';
+secondButton.id = 'secondButton';
+firstButtonAnchor.id = 'firstButtonAnchor';
+secondButtonAnchor.id = 'secondButtonAnchor';
+popupContainer1.id = 'popupContainer1'; // Background container
+popupContainer2.id = 'popupContainer2'; // Main container (with all child elements)
+cardButtonContainer.id = 'cardButtonContainer';
+popupCardButtonIcon1.id = 'popupCardButtonIcon1';
+popupCardButtonIcon2.id = 'popupCardButtonIcon2';
+
+//  Add classes and aditional characteristics
+popupParagraph.classList.add('text2');
+technologiesList.classList.add('ul-with');
+firstTechnology.classList.add('blue-li');
+secondTechnology.classList.add('blue-li');
+thirdTechnology.classList.add('blue-li');
+firstButton.classList.add('green-button');
+secondButton.classList.add('green-button');
+firstButton.innerText = 'Live version   ';
+secondButton.innerText = 'Source code   ';
+firstButtonAnchor.target = '_blank';
+secondButtonAnchor.target = '_blank';
+popupCloser.src = 'images/popupCloserIcon.svg';
+popupCardButtonIcon1.src = 'images/popupCardButtonIcon1.svg';
+popupCardButtonIcon2.src = 'images/popupCardButtonIcon2.svg';
+
+//  Assing information from objects and add them to the page
+function refreshPopup(currentObject) {
+  const [a, b, c] = currentObject.technologies;
+  projectImage.src = currentObject.image;
+  firstTechnology.innerText = a;
+  secondTechnology.innerText = b;
+  thirdTechnology.innerText = c;
+  popupTittle.innerText = currentObject.name;
+  popupParagraph.innerText = currentObject.description;
+  firstButtonAnchor.href = currentObject.live;
+  secondButtonAnchor.href = currentObject.source;
+  document.body.appendChild(popupContainer1);
+}
+
+//  Add elements to containers
+technologiesList.appendChild(firstTechnology);
+technologiesList.appendChild(secondTechnology);
+technologiesList.appendChild(thirdTechnology);
+firstButton.appendChild(popupCardButtonIcon1);
+secondButton.appendChild(popupCardButtonIcon2);
+firstButtonAnchor.appendChild(firstButton);
+secondButtonAnchor.appendChild(secondButton);
+cardButtonContainer.appendChild(firstButtonAnchor);
+cardButtonContainer.appendChild(secondButtonAnchor);
+popupContainer2.appendChild(projectImage);
+popupContainer2.appendChild(popupCloser);
+popupContainer2.appendChild(popupTittle);
+popupContainer2.appendChild(technologiesList);
+popupContainer2.appendChild(popupParagraph);
+popupContainer2.appendChild(cardButtonContainer);
+popupContainer1.appendChild(popupContainer2);
```

Comment:
- [ ] Your current implementation of the modal popup isn't followint DRY principles. You are currently creating and appending every single element of the modal. A better approach would be to have one container with all the sections that obtain the data from the object, and then gets displayed on the modal using `appendChild`.

Reply:
Thank you! very much! for your help 👍 
Can you explain to me a little more about the way I need to add the elements to the container?... When you say "to have one container" do you mean to create it from the .HTML file?

Reply:
Yes, it could be either in the `index.html` file or even in the js file, appending it to the body of the page with a class that keeps its display hidden until a click event listener toggles it. There's many ways to do it that can spare your system some resources, unlike the current one. Remember Js is browser heavy!